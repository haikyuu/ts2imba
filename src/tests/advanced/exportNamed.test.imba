import {build} from "../../index.imba"
import * as imbac from 'imba/compiler'

let tsx-code = '''
export const headers = {
  xInertia: 'x-inertia',
  xInertiaVersion: 'x-inertia-version',
  xInertiaLocation: 'x-inertia-location',
  xInertiaPartialData: 'x-inertia-partial-data',
  xInertiaPartialComponent: 'x-inertia-partial-component',
  xInertiaCurrentComponent: 'x-inertia-current-component',
};
'''

let imba-code = '''
export const headers = 
	xInertia: "x-inertia"
	xInertiaVersion: "x-inertia-version"
	xInertiaLocation: "x-inertia-location"
	xInertiaPartialData: "x-inertia-partial-data"
	xInertiaPartialComponent: "x-inertia-partial-component"
	xInertiaCurrentComponent: "x-inertia-current-component"

'''
test 'inertia example' do
	const result = await build tsx-code
	expect(result.code).toEqual(imba-code)
	const out = imbac.compile(result.code, sourceId: 'sth')