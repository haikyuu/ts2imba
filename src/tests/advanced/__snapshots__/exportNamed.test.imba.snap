// Vitest Snapshot v1

exports[`inertia example 32 1`] = `
"export const headers = 
	xInertia: \\"x-inertia\\"
	xInertiaVersion: \\"x-inertia-version\\"
	xInertiaLocation: \\"x-inertia-location\\"
	xInertiaPartialData: \\"x-inertia-partial-data\\"
	xInertiaPartialComponent: \\"x-inertia-partial-component\\"
	xInertiaCurrentComponent: \\"x-inertia-current-component\\"
"
`;
